documentation.md

```diff
@@ -0,0 +1,97 @@
+# Turbo for Django
+
+This repository aims to provide utilities for working with [Turbo](https://turbo.hotwire.dev)
+with the Django web framework. 
+
+## Turbo Streams
+Currently, the repository contains utilities for working with
+[Turbo Streams](https://turbo.hotwire.dev/handbook/streams) over Websockets, the one part of
+Turbo which requires a specific integration with the backend framework to function. In Django's
+case, this means taking advantage of [Django Channels](https://github.com/django/channels).
+
+The Django Streams integration is heavily inspired by the helpers and mixins found in
+[turbo-rails](https://github.com/hotwired/turbo-rails). There are a few steps required to get
+the integration working.
+
+### Initial Setup
+#### Django Channels
+After setting up Channels according to the [documentation](https://channels.readthedocs.io/en/stable/installation.html),
+make sure to modify the top-level `ProtocolTypeRouter` to include the `TurboStreamsConsumer`:
+
+```python
+from turbo.consumers import TurboStreamsConsumer
+application = ProtocolTypeRouter({
+  "http": AsgiHandler(),
+  "websocket": TurboStreamsConsumer.as_asgi()  # Leave off .as_asgi() if using Channels 2.x
+})
+```
+
+#### Including JavaScript
+In our `<head>` for all our templates, we need to `{% include 'turbo/head.html' %}`, which will include the Turbo libraries we need
+in order to use Turbo Drive, Frames and Streams in our app.
+
+### Model Layer
+
+Throughout these docs we'll be using these two basic models for a chat app, similar to the
+[Hotwire demo video](https://www.youtube.com/watch?v=eKY-QES1XQQ):
+
+```python
+# chat/models.py
+
+from django.db import models
+class Room(models.Model):
+    name = models.CharField(max_length=255)
+
+
+class Message(models.Model):
+    room = models.ForeignKey(Room, related_name="messages", on_delete=models.CASCADE)
+    text = models.TextField()
+```
+
+`turbo-django` provides a `BroadcastableMixin` that enables CRUD operations from a given model
+to be broadcast over websockets. To enable messages to be broadcast over websocket to users
+listening in on a particular room, we add the `BroadcastableMixin` and tell `turbo-django`
+that actions on messages should be broadcast to the related `Room` as well:
```

Comment:
I like how clear this is since it's a relatively complex subject. The `as well` at the end makes it sound like we'll get updates for both `Room` and `Messages` though, but we then set `broadcast_self = False`?

Reply:
You're right, this is confusing -- I'll remove the `as well`